src/controllers/credentials.ts

```diff
@@ -391,22 +429,25 @@ export class CredentialController {
 		const result = validationResult(request);
 		// handle error
 		if (!result.isEmpty()) {
-			return response.status(StatusCodes.BAD_REQUEST).json({ error: result.array().pop()?.msg });
+			return response.status(StatusCodes.BAD_REQUEST).json({
+				error: result.array().pop()?.msg,
+			} satisfies UnsuccesfulSuspendCredentialResponseBody);
 		}
 
 		// Get publish flag
-		const publish = request.query.publish === 'false' ? false : true;
+		const { publish } = request.query as SuspendCredentialRequestQuery;
 		// Get symmetric key
-		const symmetricKey = request.body.symmetricKey as string;
+		const { credential, symmetricKey } = request.body as SuspendCredentialRequestBody;
+
 		// Get strategy e.g. postgres or local
 		const identityServiceStrategySetup = new IdentityServiceStrategySetup(response.locals.customer.customerId);
 
 		try {
 			const result = await identityServiceStrategySetup.agent.suspendCredentials(
-				request.body.credential,
-				publish,
+				credential,
+				publish as boolean,
```

Comment:
@Eengineer1 Could you please take a look here as well. I'm not sure is it right decision in terms of TypeScript ideology. Validator should transform this parameter into `boolean` type, cause there is a call `toBoolean` but request.query is still a structure of strings. So, I check it in debug session and `publish` variable has explicit type `boolean` and it seems like ok, but anyway.

Reply:
Well tracked down and yes, I do prefer assertions, especially if there's no type narrowing involved, in which case this syntax is well suited here.